test/mailer_notifier_test.exs

```diff
@@ -34,330 +34,354 @@ defmodule MailerNotifierTest do
     end
   end
 
-  describe "Bamboo" do
-    defmodule TestRouter do
-      use Phoenix.Router
-      import Phoenix.Controller
-
-      use BoomNotifier,
-        notifier: BoomNotifier.MailNotifier.Bamboo,
-        options: [
-          mailer: Support.FakeMailer,
-          from: "me@example.com",
-          to: self(),
-          subject: "BOOM error caught"
-        ],
-        custom_data: [:assigns, :logger]
-
-      pipeline :browser do
-        plug(:accepts, ["html"])
-        plug(:save_custom_data)
+  # All mail notifiers should behave the same, so define the tests across
+  # the set of notifiers.
+  #
+  # Note: Because `use` is macro, we can't inject `mail_notifier_module`
+  #       as the `notifier` value directly, because it wont be resolved
+  #       in the correct context. Instead we can define a module attribute
+  #       which gets replaced at compile time with correct atom.
+  #
+  #       We also disable module-redefinition warning just in this context,
+  #       since we are redefining the router modules for each notifier.
+
+  notifiers = [
+    {"Bamboo", BoomNotifier.MailNotifier.Bamboo},
+    {"Swoosh", BoomNotifier.MailNotifier.Swoosh}
+  ]
+
+  for {name, mail_notifier_module} <- notifiers do
+    describe name do
+      Code.put_compiler_option(:ignore_module_conflict, true)
```

Comment:
A bit gross, could write a macro/dynamically name the modules so they don't get redefined but that's probably even more gross really? We turn conflicts back on at the ~~end of the describe.~~ end of the loop. I will tweak these to match correctly and actually reduce the scope to just the `defmodule` calls.

Reply:
Reduced scope in 355130d, just wraps the two `defmodule` calls.

Reply:
I just realized `Code.put_compiler_option` was added in Elixir 1.10 😞 

I know this is even grosser but maybe we can change this to `:elixir_config.put(:ignore_module_conflict, true)` so we can still support Elixir 1.9 (this version is still being maintained so I'd like to support it)

Let me know what do you think or if you come up with a better solution

Reply:
Fixed. The only other way I think would be defining the modules themselves and passing those into the test loop, but the modules are more than a trivial amount of code so probably best to go with the disabled warning.

Reply:
I noticed a few other calls which I fixed here too: https://github.com/wyeworks/boom/pull/71/commits/79c60ab589ddd04b83b6dc228ccd213887bad1e0

I mislabeled the other commit as targeting 1.10 instead of 1.9 but I assume you'll squash all these down anyway if it merges.